Add information about merge queues to repo rulesets guide (#1280)
semgrep/semgrep-docs: docs/kb/semgrep-ci/github-repository-rulesets-semgrep.md

The example repository is internal, so it can only be used to store workflows that run on internal and private repositories.

### Recommended configuration with merge queues

> If you use [merge queues](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/configuring-pull-request-merges/managing-a-merge-queue) for any repositories that will be scanned with this workflow, your config must include `merge_group` as a trigger in the `on:` block. Otherwise, [the workflow will not be able to run in the merge queue](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/configuring-pull-request-merges/managing-a-merge-queue#triggering-merge-group-checks-with-github-actions), and can block the queue.

Reviewer: ```suggestion
If you use [merge queues](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/configuring-pull-request-merges/managing-a-merge-queue) for repositories that will be scanned with this workflow, your config must include `merge_group` as a trigger in the `on:` block. Otherwise, [the workflow cannot run in the merge queue](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/configuring-pull-request-merges/managing-a-merge-queue#triggering-merge-group-checks-with-github-actions) and can block the queue.
```

> Unlike for `pull_request` event types, Semgrep does not have any automatic configuration to run diff scans on `merge_group` events, so additional configuration would be required to run diff scans in this environment. The most straightforward solution is to configure the workflow to be skipped during the merge group check, since the primary goal of a Semgrep diff scan is to inform the developer **before** merge whether they are introducing security issues.

Reviewer: ```suggestion
Unlike for `pull_request` event types, Semgrep does not have any automatic configuration to run diff scans on `merge_group` events, so additional configuration is needed to run diff scans in this environment. The most straightforward solution is to configure the workflow to be skipped during the merge group check, since the primary goal of a Semgrep diff scan is to inform the developer **before** merging if they are introducing security issues.
```

Author: I like the "is needed" change here, but I'm not convinced about "merging if". Weren't we just talking about preferring bare verbs over gerunds? :) I also think whether is clearer than if, but I don't feel that strongly about that part.

Reviewer: I don't feel strongly, but in my mind, before (as an adverb) should be followed by a verb and in this instance, merging seems to act as a verb while merge acts as a noun (sorry for the big nerd energy right now). I could see the opposite case being argued, so I leave it to you :)

> With the recommended alterations, and removal of event types that do not occur with repository rulesets, the [sample configuration](/docs/semgrep-ci/sample-ci-configs/#sample-github-actions-configuration-file) would look like this:

Reviewer: ```suggestion
With the recommended alterations and removal of event types that do not occur with repository rulesets, the [sample configuration](/docs/semgrep-ci/sample-ci-configs/#sample-github-actions-configuration-file) would look like this:
```
